content/ember/v6/deprecate-ember-mixin.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Deprecating Ember.Mixin
+until: 7.0.0
+since: 6.5.0
+---
+
+`Ember.Mixin` is deprecated. This is part of the legacy `@ember/object` API and is not compatible with native classes. Instead of using mixins, you should refactor your code to use class-based patterns.
```

Comment:
My strategy for removing mixins from a codebase has been to copy the code into the places that mixed in the mixin and then if there is any common state, extract that to a service. From there I refactor to utilities or to the service to clean it up.

Reply:
See also: https://github.com/tc39/proposal-decorators?tab=readme-ov-file#classes